### PR TITLE
fix specs for select option

### PR DESCRIPTION
**Pull request by smpallen99**

### CODE REVIEW

File: lib/one_model.ex [hunk 1]
```diff
@@ -163,7 +163,7 @@ defmodule OneModel do
         * Passing a list of keys will return the default schema with the values for the specified keys.
       * `limit: integer`
       """
-      @spec list(keyword()) :: [t()]
+      @spec list(keyword()) :: any
```
Comment (palexanderm) [inline lib/one_model.ex]: It's still going to return a list, so let's no drop that part of the contract.
```suggestion
      @spec list(keyword()) :: [any]
```

File: lib/one_model.ex [hunk 2]
```diff
@@ -203,7 +203,7 @@ defmodule OneModel do
           #{@schema}.list_by(field1: value1, select: :field2)
           #{@schema}.list_by(field2: value2, select: [:field1, :field3], limit: 2)
       """
-      @spec list_by(keyword()) :: [t()]
+      @spec list_by(keyword()) :: any
```
Comment (palexanderm) [inline lib/one_model.ex]: And similarly here:
```suggestion
      @spec list_by(keyword()) :: [any]
```

**Review by palexanderm — CHANGES_REQUESTED** [review]

File: lib/one_model.ex (hunk 1 above)
Comment (Sinc63) [reply lib/one_model.ex]: Or as Steve and I discussed yesterday
```suggestion
      @spec list(keyword()) :: list
```

**Review by Sinc63 — APPROVED** [review]

File: lib/one_model.ex (hunk 1 above)
Comment (smpallen99) [reply lib/one_model.ex]: I used `iist`

File: lib/one_model.ex (hunk 2 above)
Comment (smpallen99) [reply lib/one_model.ex]: went with `list`

**Pull request merged**